app/src/main/java/com/example/instagram/fragments/DetailFragment.java

```diff
@@ -0,0 +1,289 @@
+import com.example.instagram.adapters.PostsAdapter;
+import com.example.instagram.databinding.FragmentDetailBinding;
+import com.example.instagram.databinding.FragmentNewsfeedBinding;
+import com.example.instagram.models.Comment;
+import com.example.instagram.models.Post;
+import com.example.instagram.models.User;
+import com.parse.FindCallback;
+import com.parse.ParseException;
+import com.parse.ParseFile;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+
+import org.parceler.Parcels;
+import org.w3c.dom.Text;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class DetailFragment extends Fragment {
+
+    private static final String TAG = "DetailFragment";
+    private Post post;
+    private FragmentDetailBinding fragmentDetailBinding;
+
+    ImageView ivProfileImage;
+    TextView tvDescription;
+    TextView tvUsername;
+    ImageView ivImage;
+    ImageView ivLike;
+    ImageView ivComment;
+    ImageView ivShare;
+    TextView tvUsernameDescription;
+    TextView tvNumLike;
+    TextView tvNumComment;
+    TextView tvTimestamp;
+    RecyclerView rvComments;
+    CommentsAdapter adapter;
+    List<Comment> comments;
+    LinearLayoutManager linearLayoutManager;
+    EditText etComment;
+    Button btnComment;
+    User user;
+
+    public DetailFragment() {
+    }
+
+    public static DetailFragment newInstance(Parcelable post) {
+        DetailFragment fragment = new DetailFragment();
+        Bundle args = new Bundle();
+        args.putParcelable(MainActivity.KEY_POST, post);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+            post = Parcels.unwrap(getArguments().getParcelable(MainActivity.KEY_POST));
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        fragmentDetailBinding = FragmentDetailBinding.inflate(inflater, container, false);
+        return fragmentDetailBinding.getRoot();
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        ivProfileImage = fragmentDetailBinding.ivProfileImg;
+        tvUsername = fragmentDetailBinding.tvUsername;
+        tvDescription = fragmentDetailBinding.tvDescription;
+        ivImage = fragmentDetailBinding.ivImg;
+        ivLike = fragmentDetailBinding.ivLike;
+        ivComment = fragmentDetailBinding.ivComment;
+        ivShare = fragmentDetailBinding.ivShare;
+        tvTimestamp = fragmentDetailBinding.tvTimeStamp;
+        tvUsernameDescription = fragmentDetailBinding.tvUsernameDescription;
+        tvNumLike = fragmentDetailBinding.tvNumLike;
+        rvComments = fragmentDetailBinding.rvComment;
+        etComment = fragmentDetailBinding.edComment;
```

Comment:
Try to match the names with those in the XML files (lines 100, 101, 105, 108, 109)